src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler/Deprecation.php

```diff
@@ -0,0 +1,351 @@
+                $this->originatingMethod = $parsedMsg['method'];
+                // If the deprecation has been triggered via
+                // \Symfony\Bridge\PhpUnit\Legacy\SymfonyTestsListenerTrait::endTest()
+                // then we need to use the serialized information to determine
+                // if the error has been triggered from vendor code.
+                $this->internal = isset($parsedMsg['triggering_file'])
+                    && $this->pathOriginatesFromVendor($parsedMsg['triggering_file']);
+
+                return;
+            }
+            $this->originatingClass = isset($line['object']) ? \get_class($line['object']) : $line['class'];
+            $this->originatingMethod = $line['function'];
+        }
+    }
+
+    /**
+     * @return bool
+     */
+    private function lineShouldBeSkipped(array $line)
+    {
+        if (!isset($line['class'])) {
+            return true;
+        }
+        $class = $line['class'];
+
+        return 'ReflectionMethod' === $class || 0 === strpos($class, 'PHPUnit_') || 0 === strpos($class, 'PHPUnit\\');
+    }
+
+    /**
+     * @return bool
+     */
+    public function originatesFromAnObject()
+    {
+        return isset($this->originatingClass);
+    }
+
+    /**
+     * @return bool
+     */
+    public function isInternal()
+    {
+        return $this->internal;
+    }
+
+    /**
+     * @return string
+     */
+    public function originatingClass()
+    {
+        if (null === $this->originatingClass) {
+            throw new \LogicException(
+                'Check with originatesFromAnObject() before calling this method'
+            );
+        }
+
+        return $this->originatingClass;
+    }
+
+    /**
+     * @return string
+     */
+    public function originatingMethod()
+    {
+        if (null === $this->originatingMethod) {
+            throw new \LogicException(
+                'Check with originatesFromAnObject() before calling this method'
+            );
+        }
+
+        return $this->originatingMethod;
+    }
+
+    /**
+     * @param string $utilPrefix
+     *
+     * @return bool
+     */
+    public function isLegacy($utilPrefix)
+    {
+        $test = $utilPrefix.'Test';
+        $class = $this->originatingClass();
+        $method = $this->originatingMethod();
+
+        return 0 === strpos($method, 'testLegacy')
+            || 0 === strpos($method, 'provideLegacy')
+            || 0 === strpos($method, 'getLegacy')
+            || strpos($class, '\Legacy')
+            || \in_array('legacy', $test::getGroups($class, $method), true);
+    }
+
+    /**
+     * Tells whether both the calling package and the called package are vendor
+     * packages.
+     *
+     * @return bool
+     */
+    public function isIndirect()
+    {
+        $erroringFile = $erroringPackage = null;
+        foreach ($this->trace as $line) {
+            if (\in_array($line['function'], array('require', 'require_once', 'include', 'include_once'), true)) {
```

Comment:
In hindsight, not sure about this… I noticed `$line['function']` has `require` as a value when autoloading happens, so I should probably leave out the other values, right?